polaris-react/src/components/TextField/TextField.tsx

```diff
@@ -118,6 +119,8 @@ interface NonMutuallyExclusiveProps {
   role?: string;
   /** Limit increment value for numeric and date-time inputs */
   step?: number;
+  /** Hide stepper by default, reveal on hover and focus */
+  stepperShownOnInteraction?: boolean;
```

Comment:
Having it be the default behavior also mitigates conflicting UIs in different sections in the admin where some TextFields wouldn't have this prop enabled. +1 to this suggestion

Reply:
@alex-page @laurkim Thank you for your suggestions! 

I can see how we may enforce it eventually as a strong design opinion in the future. However, I'm not sure if we can do it right away without introducing breaking changes for consumers.

Conditionally hiding and showing the spinner may result in an unexpected layout jump caused by increased `textfield` width depending on _how consumers handle those text field layouts_ at the parent level. 



<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React, {useCallback, useState} from 'react';

import {
  AlphaCard,
  Columns,
  Page,
  TextField,
  Text,
  Box,
  Inline,
  AlphaStack,
} from '../src';

export function Playground() {
  const [value, setValue] = useState('1');
  const handleChange = useCallback((newValue) => setValue(newValue), []);

  return (
    <Page title="Playground">
      <AlphaStack gap="5">
        <AlphaCard>
          <Text as="h2" variant="headingMd">
            Controlled width example with 1fr `Columns` component
          </Text>
          <Box paddingBlockStart="4">
            <Columns
              columns={{
                md: '1fr 1fr',
              }}
              gap={{
                md: '2',
              }}
            >
              <TextField
                label="Quantity"
                type="number"
                value={value}
                onChange={handleChange}
                autoComplete="off"
                stepperShownOnInteraction
              />
              <TextField
                label="Minimum"
                type="number"
                value={value}
                onChange={handleChange}
                autoComplete="off"
                stepperShownOnInteraction
              />
            </Columns>
          </Box>
        </AlphaCard>
        <AlphaCard>
          <Text as="h2" variant="headingMd">
            Uncontrolled width example with `Inline` component
          </Text>
          <Box paddingBlockStart="4">
            <Inline gap="4">
              <TextField
                label="Quantity"
                type="number"
                value={value}
                onChange={handleChange}
                autoComplete="off"
                stepperShownOnInteraction
              />
              <TextField
                label="Minimum"
                type="number"
                value={value}
                onChange={handleChange}
                autoComplete="off"
                stepperShownOnInteraction
              />
            </Inline>
          </Box>
        </AlphaCard>
      </AlphaStack>
    </Page>
  );
}

```

</details>

https://user-images.githubusercontent.com/15054990/224145375-9a8f8346-441f-4488-81fc-544f94e4feb2.mp4

---

@alex-page Would we consider this change as an interim step towards global change while giving the teams a warning that soon it will become the default experience and they need to use this prop proactively in their components and adjust the layouts if needed before the next major version with breaking changes rolls out deprecating this prop as a configurable option?

Reply:
Breaking changes are good. We definitely should break more stuff.

Reply:
@zecarlostorre did mention it was something we were _considering_ as an evolution of the design language. I do think it would be a nice change given how busy a lot of these surfaces get so I'm certainly for it. Any steps needed for alignment?

Reply:
Would the following work?
- **V10.36.0**
  - Ship this PR with the ability to configure the stepper via a prop
  - Inform the teams that they should add it to all input=“number” components and update the layouts if broken to give them time to migrate and adjust properly so that there are no frustrations from teams and merchants using those UIs
- **V11.0.0**
  - Remove this prop to enable the behaviour by default - by the time this ships, the majority of the teams will have adjusted to this change so it doesn’t come in as a surprise

Reply:
The `Inline` bug above should be addressed. We don't want the input jumping like that in 1P/3P apps if they update their Polaris version.

I could see this shipping without it being a breaking change and just applying the styles to the input only. I might be missing the reason why this needs to be a breaking change. Adding a property to remove it in the next version is super confusing to developers, educating them to use something and then it being removed is probably not the best approach.

Reply:
The `<Inline>` is not a bug, it’s just an example of an _uncontrolled_ children’s width from a parent level. It could happen the same way if `<Columns>` are used with something more flexible than controlled width (e.g. `auto` or `minmax()` instead of `fr` or `percentage`) and there are a lot of other cases that don’t explicitly control width that could break the same way.

All that to say,   I’m all for the bold move and strong design opinions, just wondering who’d be handling the aftermath in case it breaks for consumers.

If we are confident it won't break - I can update the PR to enforce this behaviour by default and remove the prop.

Reply:
Fixed!